src/compiler/types.ts

```diff
@@ -1048,10 +1057,19 @@ namespace ts {
         expression: Expression;
     }
 
+    /**
+      * This interface is a base interface for ObjectLiteralExpression and JSXAttributes to extend from. JSXAttributes is similar to
+      * ObjectLiteralExpression in that it contains array of properties; however, JSXAttributes' properties can only be
+      * JSXAttribute or JSXSpreadAttribute. ObjectLiteralExpression, on the other hand, can only have properties of type
+      * ObjectLiteralElement (e.g. PropertyAssignment, ShorthandPropertyAssignment etc.)
+     **/
+    export interface ObjectLiteralExpressionBase<T extends ObjectLiteralElementLike> extends PrimaryExpression, Declaration {

```

Comment:
this name seems weird when used with its type parameter: `ObjectLiteralExpressionBase<ObjectLiteralElement>`. I would prefer a more generic name instead of the suffix `Base`, maybe `LiteralExpression`? `LiteralExpression<ObjectLiteralElement>` and `LiteralExpression<JSXAttribute | JSXSpreadAttribute>` looks more reasonable.